tdrive/backend/node/src/services/documents/services/engine/index.ts

```
@@ -17,10 +17,14 @@ export class DocumentsEngine implements Initializable {
       id: e.context.company.id,
     });
     const language = receiver.preferences?.language || "en";
-    const emailTemplate =
-      event === DocumentEvents.DOCUMENT_SAHRED
-        ? "notification-document-shared"
-        : "notification-document-version-updated";
+
+    const emailTemplate = eventToTemplateMap[event];
+
+    if (!emailTemplate) {
+      logger.error(`Unknown event type: ${event}`);
+      return; // Early return on unknown event type
```

this should throw no ? If we get an unknown event, calling code must be buggy too and interrupted no ?

---

The email notification is called through the event bus, so the calling code won't be interrupted either ways. Probably I can improve the error message here to indicate it belongs the notification dispatcher.